core/service/src/lib.rs

```diff
@@ -717,6 +719,23 @@ fn build_network_future<
 						let _ = sender.send(network_state);
 					}
 				}
+				rpc::system::Request::NodeRole(sender) => {
+					use rpc::system::NodeRole;
+
+					let role = if roles.intersects(Roles::AUTHORITY) {
```

Comment:
Sorry for a late review, just noticed it. As I said earler I think it would be best to just return all the roles instead of a single one in the RPC.

The code like this is not future proof, and if we ever happen to add more roles we will need to change the format or return incomplete data.

Also simply returning all `roles` will not require any changes in the RPC if we ever add new ones. If we want to return something human readable, I'd rather have a code that looks like this:
1. Iterate over all bits of `roles`
2. If it's `1` and known return the name of the role it corresponds to.
3. If it's `1`, but not known, return `UnknownRole(<bitnumber>)`

Reply:
Oh, right, that suggestion makes sense now, thanks!